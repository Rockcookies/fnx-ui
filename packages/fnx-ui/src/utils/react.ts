import { Children, ReactElement } from 'react';
import { isFragment } from 'react-is';

export const toElementArray = (
	children: React.ReactNode,
	option?: { keepEmpty?: boolean },
): ReactElement[] => {
	const keepEmpty = option && option.keepEmpty;
	let ret: React.ReactElement[] = [];

	Children.forEach(children, (child: any) => {
		if ((child === undefined || child === null) && !keepEmpty) {
			return;
		}

		if (Array.isArray(child)) {
			ret = ret.concat(toElementArray(child));
		} else if (isFragment(child) && child.props) {
			ret = ret.concat(toElementArray(child.props.children, option));
		} else {
			ret.push(child);
		}
	});

	return ret;
};
