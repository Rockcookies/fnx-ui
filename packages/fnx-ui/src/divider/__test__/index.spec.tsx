import React from 'react';
import { render } from '@testing-library/react';
import Divider from '../index';

describe('<Divider>', () => {
	it('should render Divider have Children correctly', () => {
		const { container } = render(
			<Divider contentPosition="right">right</Divider>,
		);

		expect(container).toMatchSnapshot();
	});

	it('hairline', () => {
		const { container } = render(<Divider hairline></Divider>);

		expect(container).toMatchSnapshot();
	});

	it('contentposition defines the position of the text', () => {
		const { container } = render(
			<Divider contentPosition="right">right</Divider>,
		);
		expect(container).toMatchSnapshot();
	});

	it('using dash props, divide the line into dashed lines', () => {
		const { container } = render(<Divider dashed />);

		expect(container).toMatchSnapshot();
	});
});
