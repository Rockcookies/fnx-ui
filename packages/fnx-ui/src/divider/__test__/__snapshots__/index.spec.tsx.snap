// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider> contentposition defines the position of the text 1`] = `
<div>
  <div
    class="fnx-divider fnx-divider--hairline fnx-divider--content-right"
  >
    right
  </div>
</div>
`;

exports[`<Divider> hairline 1`] = `
<div>
  <div
    class="fnx-divider fnx-divider--hairline"
  />
</div>
`;

exports[`<Divider> should render Divider have Children correctly 1`] = `
<div>
  <div
    class="fnx-divider fnx-divider--hairline fnx-divider--content-right"
  >
    right
  </div>
</div>
`;

exports[`<Divider> using dash props, divide the line into dashed lines 1`] = `
<div>
  <div
    class="fnx-divider fnx-divider--dashed fnx-divider--hairline"
  />
</div>
`;
